### PR TITLE
CLDR-15123 subtype: robustness, fix parsing

### PR DESCRIPTION
**Pull request by srl295**
- there was a problem parsing the document
- this fix causes such failures to not crash the parsing or startup
- fix parsing (CSS selector was off)

CLDR-15123

- [X] This PR completes the ticket.

~does not deal with root cause (something with jsoup?) but makes sure that we can start up a ST from scratch~

### CODE REVIEW

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SubtypeToURLMap.java [hunk 1]
```diff
@@ -199,11 +199,18 @@ public List<String> getList() {
...
+            URLMapReader mapReader = new URLMapReader();
+            mapReader.read(utf8Data);
+
+            this.map = mapReader.getMap();
+            this.urlList = mapReader.getList();
+        } catch(Throwable t) {
+            t.printStackTrace();
+            System.err.println("Error " + t +" constructing SubtypeToURLMap, using empty SubtypeToURLMap");
+            this.map = new HashMap<>();
+            final List<String> newList = new ArrayList<>();
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/SubtypeToURLMap.java:212]: newList seems superfluous?

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SubtypeToURLMap.java [hunk 2]
```diff
@@ -199,11 +199,18 @@ public List<String> getList() {
      * @throws IllegalArgumentException
      */
     protected SubtypeToURLMap(BufferedReader utf8Data) throws IllegalArgumentException{
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/SubtypeToURLMap.java:201]: no longer "throws IllegalArgumentException"?

**Review by btangmu — CHANGES_REQUESTED** [review]

**Comment by btangmu** [comment]
[INFO] Running org.unicode.cldr.unittest.web.TestSubtypeToURLMap
Dec 13, 2022 3:25:14 AM org.unicode.cldr.web.SubtypeToURLMap$URLMapReader handleComplete
INFO: SubtypeToURLMap: read 32 lines, 2 urls and 7 subtypes mapped.
java.lang.IllegalArgumentException: Line 25: No subtype named 'dontMixLinearAAndCyproMinoanDigitsOrThingsMightBreak'. See CheckCLDR.Subtype.
	at org.unicode.cldr.web.SubtypeToURLMap$URLMapReader.handleSubtype(SubtypeToURLMap.java:150)
	at org.unicode.cldr.web.SubtypeToURLMap$URLMapReader.handleLine(SubtypeToURLMap.java:123)
	at org.unicode.cldr.web.SubtypeToURLMap$URLMapReader.read(SubtypeToURLMap.java:93)
	at org.unicode.cldr.web.SubtypeToURLMap.<init>(SubtypeToURLMap.java:204)
	at org.unicode.cldr.web.SubtypeToURLMap.getInstance(SubtypeToURLMap.java:352)
...

[INFO] 
Error:  Failures: 
Error:    TestSubtypeToURLMap.TestFailure:80 bad maps (7 failures)
	org.opentest4j.AssertionFailedError: subtypes_bad_badtype.txt: Should fail to parse ==> Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
	org.opentest4j.AssertionFailedError: subtypes_bad_badurl.txt: Should fail to parse ==> Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
	org.opentest4j.AssertionFailedError: subtypes_bad_dangling.txt: Should fail to parse ==> Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
	org.opentest4j.AssertionFailedError: subtypes_bad_dangling2.txt: Should fail to parse ==> Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
	org.opentest4j.AssertionFailedError: subtypes_bad_noend.txt: Should fail to parse ==> Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
	org.opentest4j.AssertionFailedError: subtypes_bad_nostart.txt: Should fail to parse ==> Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
	org.opentest4j.AssertionFailedError: subtypes_bad_nourl.txt: Should fail to parse ==> Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
[INFO] 
Error:  Tests run: 772, Failures: 1, Errors: 0, Skipped: 44

**Comment by srl295** [comment]
Thanks. So the test case needs work.

**Review by macchiati — APPROVED** [review]

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SubtypeToURLMap.java (hunk 1 above)
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/SubtypeToURLMap.java:212]: good catch. that was supposed to be `this.urlList = new ArrayList…` … fixed

**Comment by srl295** [comment]
ptal

**Review by btangmu — DISMISSED** [review]

**Review by btangmu — APPROVED** [review]

**Comment by srl295** [comment]
This actually did fix the root cause, so moving to review

**Pull request merged**